docs/site/Deploying-with-pm2-and-nginx.md

```diff
@@ -0,0 +1,137 @@
+---
+lang: en
+title: 'Deploying with pm2 and nginx'
+keywords: LoopBack, pm2, nginx
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/deploying-with-pm2-and-nginx.html
+---
+
+This is a basic guide for deploying a LoopBack 4 (LB4) app using
+[pm2](https://www.npmjs.com/package/pm2) behind nginx reverse proxy.
+
+PM2 is a Production Runtime and Process Manager for Node.js applications with a
+built-in Load Balancer. It allows you to keep applications alive forever, to
+reload them without downtime and facilitate common Devops tasks.
+
+NGINX is open source software for web serving, reverse proxying, caching, load
+balancing, media streaming, and more. It started out as a web server designed
+for maximum performance and stability. In addition to its HTTP server
+capabilities, NGINX can also function as a proxy server for email (IMAP, POP3,
+and SMTP) and a reverse proxy and load balancer for HTTP, TCP, and UDP servers.
+
+**NOTE**: Production deployment is a detailed topic and explained with clarity
+in their respective documentations. Please refer [pm2 docs](https://pm2.io/doc/)
+and
+[nginx docs](https://docs.nginx.com/nginx/admin-guide/installing-nginx/installing-nginx-open-source/)
+for detailed setup instructions. This guide assumes you have nginx setup
+already.
+
+## First steps
+
+### Loopback app
+
+Before we start with deployment, let's get our app ready. If you have loopback
+cli installed, run the following command to create a new app:
+
+```sh
+$ lb4 app
+```
+
+Refer to
+[getting started](https://loopback.io/doc/en/lb4/Getting-started.html#create-a-new-project)
+section of documentation for detailed instructions.
+
+### pm2 setup
+
+> > You might find yourself in a situation in which you do not have access to
+> > the CLI to start your Node.js applications. In such a situation, pm2 must be
+> > added as a dependency and must be called with the start script.
+
+1. Generate an ecosystem.config.js template with:
+
+   ```sh
+   $ pm2 init
+   ```
+
+2. Modify the generated file to match loopback requirements:
+
+   {% include ecosystem.config.js content="/ecosystem.config.js" %}
```

Comment:
```suggestion
   {% include code-caption.html content="/ecosystem.config.js" %}
```

Reply:
I think this line is causing the build failure.

Reply:
Yes, I was not very clear from the documentation about `code-caption.html`. Ran the checks and it works fine.

Reply:
Would you mind improving our docs to make it more clear for future readers?